### ISSUE TITLE
`pod setup` does not work for Cocoapods 1.8.0

### ISSUE THREAD

**Issue by wliumelb** [flutter/flutter#41253]
I was doing a fresh Flutter installation on a new Mac.

Following the instructions of Flutter doctor to install cocoapods with `sudo gem install cocoapods`, Cocoapods 1.8.0 was installed. 
Then after running `pod setup`, nothing happened, and cocoapods was still not initialised.

The solution was to remove cocoapods with `sudo gem uninstall cocoapods` and then install Cocoapods 1.7.5 with `sudo gem install cocoapods -v 1.7.5`, then `pod setup`.

I know this is probably a Cocoapods issue, but the flutter doctor instructions need to change so new users can get Flutter to work without any trouble.

**Comment by stuartmorgan** [comment]
It looks like our initialization check is looking for the local repo, but 1.8.0 uses the CDN by default which doesn't require it, so it's likely that `pod setup` is no longer pulling it down.

We'll need to figure out how we want to handle this (assuming that is indeed what's going on). We should likely switch our Podspec template to use the CDN, but we'll need to see what is needed to handle projects with an existing Podspec (e.g., does Cocoapods now fetch it on demand when it encounters a Podspec using it?) and update the setup check accordingly, or we'll need to migrate existing Podspecs.

/cc @jmagman

**Comment by blackholegalaxy** [comment]
Same here with 1.8.1. `pod setup` does nothing. Rollbacking to 1.7.5 was a successful workaround.

**Comment by davidcavazos** [comment]
I'm also getting this issue with 1.8.1, I can also confirm installing cocoapods 1.7.5 fixes it. Make sure to uninstall previous versions or it won't work! :)

1. `sudo gem uninstall cocoapods`
1. `sudo gem install cocoapods -v 1.7.5`
1. `pod setup`
1. `flutter doctor -v`

Maybe there should be a (temporary) note in the docs in the [iOS setup](https://flutter.dev/docs/get-started/install/macos#ios-setup) while this is fixed.

**Comment by jmagman** [comment]
> We'll need to figure out how we want to handle this (assuming that is indeed what's going on). We should likely switch our Podspec template to use the CDN, but we'll need to see what is needed to handle projects with an existing Podspec (e.g., does Cocoapods now fetch it on demand when it encounters a Podspec using it?) and update the setup check accordingly, or we'll need to migrate existing Podspecs.

It uses the CDN by default as of 1.8.0.  What would need to be migrated in the Podfile (assuming you meant Podfile and not podspec)?

**Comment by jmagman** [comment]
> Maybe there should be a (temporary) note in the docs in the [iOS setup](https://flutter.dev/docs/get-started/install/macos#ios-setup) while this is fixed.

https://github.com/flutter/website/pull/3041/

**Comment by stuartmorgan** [comment]
> What would need to be migrated in the Podfile (assuming you meant Podfile and not podspec)?

Turns out nothing; I was thinking of the release notes mentioning changing the source:
```
- source 'https://github.com/CocoaPods/Specs.git'
+ source 'https://cdn.cocoapods.org/'
```
and assumed we had that in our Podfile. But I was just remembering seeing the commented-out CDN source line.

So does that mean all we need to do here is make 1.8 the minimum and remove the check that the local repo is checked out (and the corresponding instructions)?

**Comment by jmagman** [comment]
> So does that mean all we need to do here is make 1.8 the minimum and remove the check that the local repo is checked out (and the corresponding instructions)?

Check out https://github.com/flutter/flutter/pull/41491.  I skipped the local check then the version is >= 1.8.  I removed the pod setup instructions from the installer comment, but they will `pod setup` instructions if they run again with an installed, uninitialized version < 1.8.

**Comment by jeyj31** [comment]
Hi everyone, 

I'm currently trying to install flutter on my macOS but I can't install the CocoaPods even with your advices and all things I could find on internet 

someone could please help me ? 

attached, the screenshot of my terminal 

<img width="600" alt="Capture d’écran 2019-09-28 à 15 19 07" src="https://user-images.githubusercontent.com/55918714/65817076-562bc880-e203-11e9-93af-2f040420ad11.png">

**Comment by blackholegalaxy** [comment]
> Hi everyone,
> 
> I'm currently trying to install flutter on my macOS but I can't install the CocoaPods even with your advices and all things I could find on internet
> 

As stated in https://github.com/flutter/flutter/issues/41253#issuecomment-536039050 you have to uninstall any previous version of cocoapods and install specific 1.7.5 version. You also ensure you didn't have any version installed through homebrew which could interfer with the gem one.

**Comment by jeyj31** [comment]
Ok and for uninstall home-brew there is the same command ?

**Comment by blackholegalaxy** [comment]
> Ok and for uninstall home-brew there is the same command ?

`brew uninstall cocoapods`

**Comment by paulgillingwater** [comment]
> Hi everyone,
> 
> I'm currently trying to install flutter on my macOS but I can't install the CocoaPods even with your advices and all things I could find on internet
> 
> someone could please help me ?
> 
> attached, the screenshot of my terminal
> 
> <img alt="Capture d’écran 2019-09-28 à 15 19 07" width="600" src="https://user-images.githubusercontent.com/55918714/65817076-562bc880-e203-11e9-93af-2f040420ad11.png">

I had exactly the same problem and solved it as follows: (assuming you installed two versions of Cocoapod):

sudo gem uninstall cocoapods

Select gem to uninstall:
 1. cocoapods-1.7.5
 2. cocoapods-1.8.1
 3. All versions
> 2
Successfully uninstalled cocoapods-1.8.1
Smackbook:development paul$ pod setup
Setting up CocoaPods master repo
  $ /usr/bin/git clone https://github.com/CocoaPods/Specs.git --progress -- master
  Cloning into 'master'...
  remote: Enumerating objects: 299, done.        
  remote: Counting objects: 100% (299/299), done.        
  remote: Compressing objects: 100% (285/285), done.

**Comment by LonelyCpp** [comment]
I have cocoapods 8.0 installed, and flutter doctor complains about it. But the iOS builds are not affected by it. Looks like this can be ignored rather than downgrading to v1.7.5

**Comment by coldix** [comment]
Thanks Paul, I had the same problem, this fixed it :)
![cocoapods-fix](https://user-images.githubusercontent.com/38780650/66249425-9c8aa580-e776-11e9-84e2-1156cd693426.png)

**Comment by davidair** [comment]
Folks, please avoid downgrading CocoaPods to work around a "flutter doctor" issue - CocoaPods 1.8+ usage of CDN is a fantastic feature and an outstanding performance optimization. Unless something is broken, try using the latest version (also, https://github.com/flutter/flutter/pull/41491 has the fix for Flutter).

**Comment by jmagman** [comment]
> Folks, please avoid downgrading CocoaPods to work around a "flutter doctor" issue - CocoaPods 1.8+ usage of CDN is a fantastic feature and an outstanding performance optimization. Unless something is broken, try using the latest version (also, #41491 has the fix for Flutter).

It's not just a `flutter doctor` issue, it's a `flutter build`/`run` issue.  When the local repo isn't present `pod install` [is skipped](https://github.com/flutter/flutter/blob/master/packages/flutter_tools/lib/src/macos/cocoapods.dart#L195).  https://github.com/flutter/flutter/commit/1eb1e2226bef1121103d5659951e297623177ffc is in v1.10.7, so once that hits stable the [`-v 1.7.5` instruction and note](https://flutter.dev/docs/get-started/install/macos#deploy-to-ios-devices) can be removed.

**Comment by blackholegalaxy** [comment]
Since the MR to fix that on 1.8+ detection, maybe it could be useful to include that as hotfix in current/next stable.

**Comment by superHS** [comment]
thanks,,,i fixed it

**Comment by jmagman** [comment]
> Since the MR to fix that on 1.8+ detection, maybe it could be useful to include that as hotfix in current/next stable.

We're evaluating hotfixing this to stable.

**Comment by kwight** [comment]
If people are feeling confident, another option is to run the `dev` [channel](https://github.com/flutter/flutter/wiki/Flutter-build-release-channels) for now, which has received the fix in #41491 mentioned by @davidair .

```bash
flutter channel dev
flutter upgrade
```

**Comment by samchang8826** [comment]
> If people are feeling confident, another option is to run the `dev` [channel](https://github.com/flutter/flutter/wiki/Flutter-build-release-channels) for now, which has received the fix in #41491 mentioned by @davidair .
> 
> ```shell
> flutter channel dev
> flutter upgrade
> ```

After I switch to dev branch, I got a lots of issues
so my suggestion is create fake folder under ~/.cocoapods/respos/master , it's fast and easy solution.

**Comment by jmagman** [comment]
This fix is now in stable as v1.9.1+hotfix.5.
